### PR TITLE
Use spring web client instead of java http client

### PR DESCRIPTION
**Pull request by IvanBorislavovDimitrov**

### CODE REVIEW

File: src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java [hunk 1]
```diff
@@ -103,18 +89,37 @@ private URI buildGetLogsUrl(UUID applicationGuid, LocalDateTime offset) {
...
+    private InputStream getResponseBodyAsStream(URI logsUrl) {
+        List<InputStream> responseStreams = WEB_CLIENT.get()
+                                                      .uri(logsUrl)
+                                                      .headers(httpHeaders -> httpHeaders.addAll(getAdditionalRequestHeaders(requestTags)))
+                                                      .header(HttpHeaders.AUTHORIZATION, oAuthClient.getAuthorizationHeaderValue())
+                                                      .exchangeToFlux(this::handleClientResponse)
+                                                      .map(DataBuffer::asInputStream)
+                                                      .collectList()
+                                                      .block();
+        return new SequenceInputStream(Collections.enumeration(responseStreams));
```
Comment (radito3) [inline src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java:101]: Are these streams closed anywhere? We should be careful about a resource leak

File: src/main/java/com/sap/cloudfoundry/client/facade/adapters/CloudFoundryClientFactory.java [hunk 2]
```diff
@@ -165,4 +169,15 @@ private reactor.netty.http.client.HttpClient getAdditionalHttpClientConfiguratio
         return clientWithOptions;
     }
 
+    private static WebClient buildWebClient() {
```
Comment (radito3) [inline src/main/java/com/sap/cloudfoundry/client/facade/adapters/CloudFoundryClientFactory.java:172]: I think the main reason for choosing the jdk http client is that the direct memory was abnormally high when using reactor netty. Does the spring client cause the same issue?

File: src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java (hunk 1 above)
Comment (IvanBorislavovDimitrov) [reply src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java]: Removed in the latest change. When closing the SequenceInputStream, it closes the streams that it is comprised of, at least according to doc:
_If this stream was created from an enumeration, all remaining elements are requested from the enumeration and closed before the close method returns._

File: src/main/java/com/sap/cloudfoundry/client/facade/adapters/CloudFoundryClientFactory.java (hunk 2 above)
Comment (IvanBorislavovDimitrov) [reply src/main/java/com/sap/cloudfoundry/client/facade/adapters/CloudFoundryClientFactory.java:179]: That's correct when using DataBuffers (ByteBuffers) to convert the response to InputStream. (Reproduced)
Parsing it from the response does not use direct memory. In any case, we're reading the whole stream to parse the response

File: src/main/java/com/sap/cloudfoundry/client/facade/adapters/CloudFoundryClientFactory.java [hunk 3]
```diff
@@ -25,31 +21,38 @@
...
-    static final HttpClient HTTP_CLIENT = HttpClient.newBuilder()
-                                                    .executor(Executors.newSingleThreadExecutor())
-                                                    .followRedirects(HttpClient.Redirect.NORMAL)
-                                                    .connectTimeout(Duration.ofMinutes(10))
-                                                    .build();
-
     private final Map<String, ConnectionContext> connectionContextCache = new ConcurrentHashMap<>();
 
+    private static final ObjectMapper MAPPER = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+                                                                 .configure(DeserializationFeature.UNWRAP_ROOT_VALUE, true);
```
Comment (radito3) [inline src/main/java/com/sap/cloudfoundry/client/facade/adapters/CloudFoundryClientFactory.java:52]: Does the unwrapping of the root value break the cf root response parsing? Or is this mapper not used when the response is interpreted as a String?

**Review by radito3 — APPROVED** [review]

File: src/main/java/com/sap/cloudfoundry/client/facade/adapters/CloudFoundryClientFactory.java (hunk 3 above)
Comment (IvanBorislavovDimitrov) [reply src/main/java/com/sap/cloudfoundry/client/facade/adapters/CloudFoundryClientFactory.java:52]: No, the mapper and the decoder are not used when parsing Strings. It is only used when parsing the response to the respective java class (application logs response)

**Pull request merged**